eth/transactionManager_test.go

```diff
@@ -0,0 +1,320 @@
+	)
+	eth.err["TransactionByHash"] = nil
+
+	// Replacement gas price exceeds max gas price
+	// Throw error
+	tx, err = tm.replace(stubTx)
+	assert.Nil(tx)
+	assert.EqualError(
+		err,
+		fmt.Sprintf("Replacement gas price exceeds max gas price suggested=%v max=%v", calcReplacementGasPrice(stubTx), gpm.maxGasPrice),
+	)
+
+	// Error signing replacement tx
+	expErr = errors.New("SignTx error")
+	sig := &stubTransactionSigner{
+		err: nil,
+	}
+	tm.sig = sig
+	sig.err = expErr
+	gpm.maxGasPrice = big.NewInt(99999)
+	tx, err = tm.replace(stubTx)
+	assert.Nil(tx)
+	assert.EqualError(err, expErr.Error())
+	sig.err = nil
+
+	// Error sending replacement tx
+	expErr = errors.New("SendTx error")
+	eth.err["SendTransaction"] = expErr
+	logsBefore := glog.Stats.Info.Lines()
+	tx, err = tm.replace(stubTx)
+	logsAfter := glog.Stats.Info.Lines()
+	assert.EqualError(err, expErr.Error())
+	assert.Equal(logsAfter-logsBefore, int64(1))
+	eth.err["SendTransaction"] = nil
+
+	// Success
+	logsBefore = glog.Stats.Info.Lines()
+	tx, err = tm.replace(stubTx)
+	logsAfter = glog.Stats.Info.Lines()
+	assert.Nil(err)
+	expTx := types.NewTransaction(1, *stubTx.To(), stubTx.Value(), 100000, calcReplacementGasPrice(stubTx), stubTx.Data())
+	assert.Equal(tx, expTx)
+	assert.Equal(logsAfter-logsBefore, int64(1))
+
+	// Replacement gas price lower than suggest gas price
+	// Use market gas price
+	gpm.gasPrice = big.NewInt(999)
+	logsBefore = glog.Stats.Info.Lines()
+	tx, err = tm.replace(stubTx)
+	logsAfter = glog.Stats.Info.Lines()
+	assert.Nil(err)
+	expTx = types.NewTransaction(1, *stubTx.To(), stubTx.Value(), 100000, gpm.gasPrice, stubTx.Data())
+	assert.Equal(tx, expTx)
+	assert.Equal(logsAfter-logsBefore, int64(1))
+}
+
+func TestTransactionManager_CheckTxLoop(t *testing.T) {
+	assert := assert.New(t)
+
+	eth := &stubTransactionSenderReader{
+		err: make(map[string]error),
+	}
+	q := transactionQueue{}
+	gasPrice := big.NewInt(10)
+	gpm := &GasPriceMonitor{
+		minGasPrice: big.NewInt(0),
+		maxGasPrice: big.NewInt(99999999),
+		gasPrice:    big.NewInt(1),
+	}
+	sig := &stubTransactionSigner{
+		err: nil,
+	}
+
+	tm := &TransactionManager{
+		maxReplacements: 1,
+		cond:            sync.NewCond(&sync.Mutex{}),
+		eth:             eth,
+		queue:           q,
+		txTimeout:       2 * time.Second,
+		gpm:             gpm,
+		sig:             sig,
+		quit:            make(chan struct{}),
+	}
+
+	eth.pending = true
+	receipt := types.NewReceipt(pm.RandHash().Bytes(), false, 100000)
+	eth.receipt = receipt
+
+	stubTx := types.NewTransaction(1, pm.RandAddress(), big.NewInt(100), 100000, gasPrice, pm.RandBytes(68))
+
+	go tm.Start()
+	defer tm.Stop()
+
+	sink := make(chan *transactionReceipt, 10)
+	sub := tm.Subscribe(sink)
+	defer sub.Unsubscribe()
+	tm.SendTransaction(context.Background(), stubTx)
+
+	event := <-sink
+	assert.NotNil(event)
+	assert.Nil(event.err)
+
+	// Wait error no replacements
```

Comment:
Isn't maxReplacement in the TransactionManager still set to 1 here? And in that case, we get a context.DeadlineExceeded error after 2 calls to wait() with the second being for a replacement tx.